### PR TITLE
Jesse/rocket control functionality

### PR DESCRIPTION
**Pull request by Jgerbrandt**

### CODE REVIEW

File: AvionicsSoftware.ioc [hunk 1]
```diff
@@ -165,46 +193,74 @@ PA15.GPIO_Label=BUZZER_PWM
...
 PA7.Locked=true
 PA7.Mode=Full_Duplex_Master
 PA7.Signal=SPI1_MOSI
 PA8.GPIOParameters=GPIO_Label
-PA8.GPIO_Label=LAUNCH
+PA8.GPIO_Label=BAT_EN
 PA8.Locked=true
 PA8.Signal=GPIO_Output
+PA9.GPIOParameters=GPIO_Label
+PA9.GPIO_Label=RCU_RS485_USART1_TX
```
Comment (AgamAulakh) [inline AvionicsSoftware.ioc]: hard to comment on an IOC file, but I think we should not add "RS485" to the pin names. To the STM, these are used as UART lines and firmware doesn't care if they're converted to a specific kind of differential pair comm

File: AvionicsSoftware.ioc [hunk 2]
```diff
@@ -165,46 +193,74 @@ PA15.GPIO_Label=BUZZER_PWM
...
+PA9.Locked=true
+PA9.Mode=Asynchronous
+PA9.Signal=USART1_TX
+PB1.GPIOParameters=GPIO_Label
+PB1.GPIO_Label=PS1_ADC1_IN9
```
Comment (AgamAulakh) [inline AvionicsSoftware.ioc]: nitpick: I don't think we should include the specific pin type in the names. When we use the user-defined names in our code, we should be blind to, for example, the actual uart line/adc channel being used. I suggest we remove the actual pin name that's kept in the middle (like changing GPS_UART4_RX to just GPS_RX). I'm definitely open to discussion on this point!

Comment (AgamAulakh) [inline AvionicsSoftware.ioc]: Although the schematic has this as "PS1," I think we should just call it AV_Pressure or something along those lines

File: AvionicsSoftware.ioc [hunk 3]
```diff
@@ -147,13 +164,24 @@ NVIC.UART5_IRQn=true\:5\:0\:false\:false\:true\:true\:true\:true\:true
...
 NVIC.UsageFault_IRQn=true\:0\:0\:false\:false\:true\:false\:false\:false\:false
 PA0-WKUP.GPIOParameters=GPIO_Label
-PA0-WKUP.GPIO_Label=GPS_UART_TX
+PA0-WKUP.GPIO_Label=GPS_UART4_TX
+PA0-WKUP.Locked=true
 PA0-WKUP.Mode=Asynchronous
 PA0-WKUP.Signal=UART4_TX
 PA1.GPIOParameters=GPIO_Label
-PA1.GPIO_Label=GPS_UART_RX
+PA1.GPIO_Label=GPS_UART4_RX
```
Comment (AgamAulakh) [inline AvionicsSoftware.ioc]: Note to Chris: GPS uart has been configured to use DMA (GPS --> mcu memory)

File: AvionicsSoftware.ioc [hunk 4]
```diff
@@ -165,46 +193,74 @@ PA15.GPIO_Label=BUZZER_PWM
...
 PA6.Mode=Full_Duplex_Master
 PA6.Signal=SPI1_MISO
 PA7.GPIOParameters=GPIO_Label
-PA7.GPIO_Label=IMU_SPI_MOSI
+PA7.GPIO_Label=IMU_SPI1_MOSI
 PA7.Locked=true
 PA7.Mode=Full_Duplex_Master
 PA7.Signal=SPI1_MOSI
 PA8.GPIOParameters=GPIO_Label
-PA8.GPIO_Label=LAUNCH
```
Comment (AgamAulakh) [inline AvionicsSoftware.ioc]: We should configure the gpio _input_ pins (LAUNCH and DMB_ABORT) to be _pulled down_

Comment (AgamAulakh) [reply AvionicsSoftware.ioc]: AUX_2 should be pull up. Though we don't currently use this pin, in the past we have used the auxiliary pins for debugging by holding them to ground.

File: Components/FlightControl/RocketSM.cpp [hunk 5]
```diff
@@ -155,24 +153,23 @@ RocketState PreLaunch::HandleNonIgnitionCommands(RocketControlCommands rcAction,
         // Transition to abort state
         return RS_ABORT;
```
Comment (AgamAulakh) [inline Components/FlightControl/RocketSM.cpp:190]: This doesn't have to be included in this PR, but we should think about making the LAUNCH and DMB_ABORT input pins trigger an interrupt in order to immediately queue the right state transition command.

File: Components/FlightControl/RocketSM.cpp [hunk 6]
```diff
@@ -348,10 +345,12 @@ RocketState Arm::HandleCommand(Command& cm)
...
+            PowerSelect::UmbilicalPower();
+            //TODO: we should check to make sure umbilical power is available before doing so
             break;
         case RSC_POWER_TRANSITION_ONBOARD:
-            //TODO: Transition to onboard power
+            PowerSelect::InternalPower();
+            //TODO: we should check to make sure internal power is available before doing so
             break;
         case RSC_GOTO_IGNITION:
             // Transition to ready for ignition state
```
Comment (AgamAulakh) [inline Components/FlightControl/RocketSM.cpp:400]: Got some more clarification on the ignition state transitions from Mo:
As soon as we enter the ignition state, we should start a timer for the igniter delay. If the timer expires before the launch command is received, we should go into Arm. If the launch command is received, we should wait until the timer expires before entering launch and opening the MEV

**Review by AgamAulakh — CHANGES_REQUESTED** [review]
Great job! Most of these comments are discussion points

**Review by cjchanx — CHANGES_REQUESTED** [review]
Last commit has modified alot of files unintended, likely due to autocrlf and Macs leaving a .DS_Store file.

Run `git config --global core.autocrlf true` on any Windows desktops to keep consistent line endings.

File: Components/FlightControl/Inc/RocketSM.hpp [hunk 7]
```diff
@@ -53,6 +53,8 @@ enum RocketState
  */
 enum RocketControlCommands
 {
+    //open vent and drain at the same time
+
```
Comment (cjchanx) [inline Components/FlightControl/Inc/RocketSM.hpp]: Remove these two lines

File: Core/.DS_Store
Comment (cjchanx) [inline Core/.DS_Store:1]: Remove :)

File: Components/FlightControl/TelemetryTask.cpp [hunk 8]
```diff
@@ -96,3 +96,23 @@ void TelemetryTask::RunLogSequence()
     //TODO: Commented out for now, until merged with the flight task changes
```
Comment (cjchanx) [inline Components/FlightControl/TelemetryTask.cpp:97]: Call this function in RunLogSequence

**Review by cjchanx — APPROVED** [review]

**Pull request merged**